save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/UsersController.kt

```diff
@@ -44,11 +47,12 @@ class UsersController(
     @PostMapping("/new")
     fun saveNewUser(@RequestBody user: User) {
         val userName = requireNotNull(user.name) { "Provided user $user doesn't have a name" }
-        userRepository.findByName(userName)?.let {
+        originalLoginRepository.findByName(userName)?.user?.let {
```

Comment:
Why are you checking `originalLoginRepository` here? What if we need to save a user with a name that exists in original logins but from a new authorization source?

Reply:
This is checked further. We will open a registration page and ask the user to change their login.

Reply:
I can see that you are trying to retrieve a user by their original username and if it's not found you start saving the user as a new user. What I'm asking is, suppose we have a user `[name = "johndoe", source = "github"]`, and then another user, `[name = "johndoe", source = "facebook"]` tries to authenticate. Seems like with current logic you would treat the new user as if they were already present, because there is a row with the same name in `original_login` table

Reply:
Ok. Agree. Changed on _findByNameAndSource_